pkg/segment/writer/segmetarw.go

```diff
@@ -505,9 +506,22 @@ func BackFillPQSSegmetaEntry(segkey string, newpqid string) {
 }
 
 // AddToBackFillAndEmptyPQSChan adds a new pqid to the channel
-// if emptyPqs is true, then it will also add the EmptyResults for this pqid
-func AddToBackFillAndEmptyPQSChan(segkey string, newpqid string, emptyPqs bool) {
-	pqsChan <- PQSChanMeta{pqid: newpqid, segKey: segkey, emptyPqs: emptyPqs}
+// Adds the pqid to the segfullmeta file for the given segkey
+// if writeToEmptyPQIDMeta is true, then it will write the EmptyResults for this pqid
+func AddToBackFillAndEmptyPQSChan(segkey string, newpqid string, writeToEmptyPQIDMeta bool) {
+	pqsChan <- PQSChanMeta{pqid: newpqid, segKey: segkey, writeToSegFullMeta: true, writeToEmptyPQIDMeta: writeToEmptyPQIDMeta}
+}
+
+// AddEmptyPQIDToChan adds a new pqid to the channel
+// if writeToEmptyPQIDMeta is true, then it will write the EmptyResults for this pqid
+func AddEmptyPQIDToChan(pqid string, segKey string) {
+	pqsChan <- PQSChanMeta{pqid: pqid, segKey: segKey, writeToEmptyPQIDMeta: true}
+}
+
+func AddEmptyPQIDMapToChan(pqid string, pqidToEmptySegMap map[string]bool) {
+	for segKey := range pqidToEmptySegMap {
+		AddEmptyPQIDToChan(pqid, segKey)
+	}
```

Comment:
This function is being called from `deleteSegmentsFromEmptyPqMetaFiles` in `retention.go` file. Previously function is directly calling`DeleteSegmentFromPqid` and now it calls this with the updated PQID Map.

Is this fine? Or this needs to be immediate?